File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
            
        def dfs(root_node):
            if not root_node:
                return True
                
            q = deque()
            q.append((root_node.left, root_node.right))
            
            while q:
                left, right = q.popleft()
                
               
                if left is None and right is None:
                    continue
                
               
                if left is None or right is None:
                    return False
                
                
                if left.val != right.val:
                    return False
                
               
                q.append((left.left, right.right))   
                q.append((left.right, right.left))  
                
            return True
            
        return dfs(root)
    

File: 0101-symmetric-tree/0101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    if(!root)return true

    const queue= [[root.left,root.right]]

    while (queue.length>0){
        const [left,right] =queue.shift()

        if (left === null && right === null) continue

        if(left==null || right===null) return false
        // if(left.left && !right.right) return false
        // if(left.right && !right.left) return false
        
      
        if(left.val != right.val){
            return false
        }
      

        queue.push([left.left,right.right]) 
       queue.push([left.right,right.left]) 

       
        
    }
     
    return true
};